### PR TITLE
CI - Build Reanimated on Windows

### PR DESCRIPTION
**Pull request by piaskowyk**
## Description

CI to test build app with Reanimated on Windows operation system.

### CODE REVIEW

**Comment by piaskowyk** [comment]
Current building results with different React Native versions - https://github.com/piaskowyk/reanimated-rn-version-tester/actions/runs/3431957313

File: .github/workflows/build-on-windows.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Test build on Windows
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/build-on-windows.yml'
+      - RNReanimated.podspec
+      - android/build.gradle
```
Comment (tomekzaw) [inline .github/workflows/build-on-windows.yml:7]:
```suggestion
      - 'android/**'
      - 'Common/**'
```

**Review by tomekzaw — APPROVED** [review]

File: .github/workflows/build-on-windows.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      - package.json
+  push:
+    branches:
+      - main
+
+env:
+  REACT_NATIVE_VERSION: "0.70.5"
```
Comment (tomekzaw) [inline .github/workflows/build-on-windows.yml:14]: What about using `latest` instead? This should work automatically for patch versions and we will know earlier if next minor version breaks something.

File: .github/workflows/build-on-windows.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    concurrency:
+      group: build-on-windows-${{ github.ref }}
+      cancel-in-progress: true
+    steps:
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: '11'
+      - name: Create React Native app
+        run: npx react-native init app --version ${{ env.REACT_NATIVE_VERSION }}
```
Comment (tomekzaw) [inline .github/workflows/build-on-windows.yml:28]: Nothing important, but there's a convention that we name all apps with PascalCase, i.e. `Example`, `FabricExample`, `WebExample`, so let's use `App` or `MyApp` here 😄 First I was thinking of `WindowsExample` but this is an Android app, just built in Windows OS.
```suggestion
        run: npx react-native init MyApp --version ${{ env.REACT_NATIVE_VERSION }}
```

File: .github/workflows/build-on-windows.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      - name: Install Reanimated
```
Comment (tomekzaw) [inline .github/workflows/build-on-windows.yml:29]: We can specify that we install the package from main branch.
```suggestion
      - name: Install Reanimated from main
```

File: .github/workflows/build-on-windows.yml (hunk 1 above)
Comment (piaskowyk) [reply .github/workflows/build-on-windows.yml]: good idea, thanks!

**Pull request merged**